idanalysis/eqparams.py

```diff
@@ -0,0 +1,1294 @@
+    def etax(self):
+        """Horizontal dispersion generated by ID.
+
+        Returns:
+            Numpy 1d array: Horizontal dispersion function
+        """
+        return self._etax
+
+    @property
+    def etapx(self):
+        """d(etax)/ds generated by ID.
+
+        Returns:
+            Numpy 1d array: Derivative of horizontal dispersion function
+        """
+        return self._etapx
+
+    @property
+    def etay(self):
+        """Vertical dispersion generated by ID.
+
+        Returns:
+            Numpy 1d array: Vertical dispersion function
+        """
+        return self._etay
+
+    @property
+    def etapy(self):
+        """d(etay)/ds generated by ID.
+
+        Returns:
+            Numpy 1d array: Derivative of vertical dispersion function
+        """
+        return self._etapy
+
+    @property
+    def i1x(self):
+        """Contribution of ID to first rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i1x
+        """
+        return self._i1x
+
+    @property
+    def i2x(self):
+        """Contribution of ID to second rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i2x
+        """
+        return self._i2x
+
+    @property
+    def i3x(self):
+        """Contribution of ID to third rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i3x
+        """
+        return self._i3x
+
+    @property
+    def i4x(self):
+        """Contribution of ID to fourth rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i4x
+        """
+        return self._i4x
+
+    @property
+    def i5x(self):
+        """Contribution of ID to fifth rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i5x
+        """
+        return self._i5x
+
+    @property
+    def i1y(self):
+        """Contribution of ID to first rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i1y
+        """
+        return self._i1y
+
+    @property
+    def i2y(self):
+        """Contribution of ID to second rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i2y
+        """
+        return self._i2y
+
+    @property
+    def i3y(self):
+        """Contribution of ID to third rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i3y
+        """
+        return self._i3y
+
+    @property
+    def i4y(self):
+        """Contribution of ID to fourth rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i4y
+        """
+        return self._i4y
+
+    @property
+    def i5y(self):
+        """Contribution of ID to fifth rad integral.
+
+        Returns:
+            Numpy 1d array: Delta i5y
+        """
+        return self._i5y
+
+    @property
+    def u0(self):
+        """Contribution of ID to energy loss.
+
+        Returns:
+            float: Energy loss per turn in ID.
+        """
+        return self._u0
+
+    @staticmethod
+    def _generate_field(a, peak, period, nr_periods, pts_period):
```

Comment:
Try to organize the code so that the hidden, auxiliary, methods come after the interface of the class. This helps when someone wants to study/maintain your code;